L1Trigger/TrackFindingTracklet/test/L1TrackNtupleMaker.cc

```diff
@@ -998,32 +1003,64 @@ void L1TrackNtupleMaker::analyze(const edm::Event& iEvent, const edm::EventSetup
         else
           myFake = 1;
 
-        myTP_pdgid = my_tp->pdgId();
-        myTP_pt = my_tp->p4().pt();
-        myTP_eta = my_tp->p4().eta();
-        myTP_phi = my_tp->p4().phi();
-        myTP_z0 = my_tp->vertex().z();
+        tmp_matchtp_pdgid = my_tp->pdgId();
+        tmp_matchtp_pt = my_tp->pt();
+        tmp_matchtp_eta = my_tp->eta();
+        tmp_matchtp_phi = my_tp->phi();
 
-        float myTP_x0 = my_tp->vertex().x();
-        float myTP_y0 = my_tp->vertex().y();
-        myTP_dxy = sqrt(myTP_x0 * myTP_x0 + myTP_y0 * myTP_y0);
+	float tmp_matchtp_vz = my_tp->vz();
+	float tmp_matchtp_vx = my_tp->vx();
+	float tmp_matchtp_vy = my_tp->vy();
+        tmp_matchtp_dxy = sqrt(tmp_matchtp_vx * tmp_matchtp_vx + tmp_matchtp_vy * tmp_matchtp_vy);
+
+	// ----------------------------------------------------------------------------------------------
+	// get d0/z0 propagated back to the IP
+
+	float tmp_matchtp_t = tan(2.0 * atan(1.0) - 2.0 * atan(exp(-tmp_matchtp_eta)));
+
+	float delx = -tmp_matchtp_vx;
+	float dely = -tmp_matchtp_vy;
+
+	float A = 0.01 * 0.5696;
```

Comment:
I changed this to now depend on the magnetic field and speed of light constants in cmssw. I also redefined the K variable to r2_inv to avoid confusion with this variable in the future. I double checked to make sure that the K and r2_inv matched after the switch to using the cms constants.